server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java

```diff
@@ -149,13 +147,20 @@ public Trigger getTrigger(TriggerKey key) throws SchedulerException {
         return scheduler.getTrigger(key);
     }
 
-    /**
-     * TODO 看起来是 yaobin 专门写给单测用的，后面优化
-     */
-    public void addTriggerListener(TriggerKey key, TriggerListener triggerListener) throws SchedulerException {
-        Matcher<TriggerKey> matcher = KeyMatcher.keyEquals(key);
-        scheduler.getListenerManager().addTriggerListener(triggerListener, matcher);
-        scheduler.getListenerManager().removeJobListener("ODC_JOB_LISTENER");
+    public CreateQuartzJobReq getQuartzJobReqOfJob(JobKey key) throws SchedulerException {
+        JobDetail jobDetail = scheduler.getJobDetail(key);
+        CreateQuartzJobReq req = new CreateQuartzJobReq();
+        req.setJobDataMap(jobDetail.getJobDataMap());
+        req.setType(JobType.valueOf(key.getGroup()));
+        req.setScheduleId(Long.parseLong(key.getName()));
+        return req;
+    }
+
+    public void updateJob(CreateQuartzJobReq req) throws SchedulerException {
```

Comment:
maybe you can use ScheduleService#update method to update the job parameters.

Reply:
@see updateTriggerDataMap#updateTriggerDataMap , in this method update trigger data map